lfs/gitscanner.go

```diff
@@ -0,0 +1,170 @@
+package lfs
+
+import (
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/rubyist/tracerx"
+)
+
+// GitScanner scans objects in a Git repository for LFS pointers.
+type GitScanner struct {
+	remote      string
+	skippedRefs []string
+
+	closed  bool
+	started time.Time
+	mu      sync.Mutex
+}
+
+// NewGitScanner initializes a *GitScanner for a Git repository in the current
+// working directory.
+func NewGitScanner() *GitScanner {
+	return &GitScanner{started: time.Now()}
+}
+
+// Close stops exits once all processing has stopped, and all resources are
+// tracked and cleaned up.
+func (s *GitScanner) Close() {
+	s.mu.Lock()
+	defer s.mu.Unlock()
+
+	if s.closed {
```

Comment:
I think you can remove the mutex lock and unlock here by using a `uint32` and either storing `0` or `1` by using the `sync/atomic` package.